patches/minecraft/net/minecraft/entity/item/EntityXPOrb.java.patch

```diff
@@ -19,11 +19,33 @@
          if (this.func_180431_b(p_70097_1_))
          {
              return false;
-@@ -208,6 +211,7 @@
+@@ -208,14 +211,16 @@
          {
              if (this.field_70532_c == 0 && p_70100_1_.field_71090_bL == 0)
              {
 +                if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.entity.player.PlayerPickupXpEvent(p_70100_1_, this))) return;
                  p_70100_1_.field_71090_bL = 2;
                  p_70100_1_.func_71001_a(this, 1);
                  ItemStack itemstack = EnchantmentHelper.func_92099_a(Enchantments.field_185296_A, p_70100_1_);
+ 
+                 if (!itemstack.func_190926_b() && itemstack.func_77951_h())
+                 {
+-                    int i = Math.min(this.func_184514_c(this.field_70530_e), itemstack.func_77952_i());
+-                    this.field_70530_e -= this.func_184515_b(i);
++                    float ratio = itemstack.func_77973_b().getXpRepairRatio(itemstack);
++                    int i = Math.min(roundAverage(this.field_70530_e * ratio), itemstack.func_77952_i());
++                    this.field_70530_e -= roundAverage(i / ratio);
```

Comment:
Potential divide by zero problem here if an item uses 0 for the xp repair ratio (to make it unrepairable by mending).

Reply:
This is definitely a problem.

Reply:
Yeah, div/0 makes this a non-starter.